Model/ProcessOrder/Cancel.php

```diff
@@ -128,9 +130,17 @@ public function execute(OrderInterface $order, array $rvvupData, string $origin)
         $processOrderResult = $this->processOrderResultFactory->create();
         $processOrderResult->setResultType(ProcessOrderResultInterface::RESULT_TYPE_ERROR);
         $processOrderResult->setRedirectPath(In::FAILURE);
-        $message = 'Payment ' . ucfirst(mb_strtolower($rvvupData['payments'][0]['status']));
-        $processOrderResult->setCustomerMessage($message);
+        if ($payment['status'] == 'DECLINED') {
+            if ($payment['declineReason'] = 'INVALID_AUTHORIZATION') {
+                $message = 'Total order value exceeds amount authorised on PayPal, ' . PHP_EOL;
```

Comment:
We should also check if this order is PayPal order, if it isn't then the error message is not correct. So can we do a check if its Paypal?

Reply:
Done